src/razor/src/document/razorDocumentSynchronizer.ts

```diff
@@ -119,6 +119,8 @@ export class RazorDocumentSynchronizer {
     }
 
     private removeSynchronization(context: SynchronizationContext) {
+        context.dispose();
+
         const documentKey = getUriPath(context.projectedDocument.uri);
         const synchronizations = this.synchronizations[documentKey];
         clearTimeout(context.timeoutId);
```

Comment:
Nit: Consider moving this inside `context.Dispose()` now that it exists?

Reply:
Since integration test have been flaky for me, I take your suggestions as a follow up and get this merged for snap.